### PR TITLE
Task 07: OrderRepository

### PR DESCRIPTION
**Pull request by KaplenkoAnastasiia**

### CODE REVIEW

File: tasks/task07/task07.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      self.items.append(order)
+    
+    def get(self, order_id: uuid.UUID):
+      for order in self.items:
+        if order.order_id == order_id:
+          return order
+        return None
+    
+    def delete(self, order_id: uuid.UUID):
+       for order in self.items:
```
Comment (ahodkov) [inline tasks/task07/task07.py]: You have `get` method for that

File: tasks/task07/task07.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      price = 0
+      if self.Goods is None or len(self.Goods.items()) < 1:
+        return price
+      for k,v  in self.Goods.items():
+        if not type(k) is Good:
+          return price
+        price += v
+      return price
+    
+    def AddGood(self,good : Good):
```
Comment (ahodkov) [inline tasks/task07/task07.py]: `@price.setter
    def price(self, value):
        self._price = value`

File: tasks/task07/task07.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      self.id = uuid.uuid4()
+      self.price = price
+
+
+class Order():
+    def __init__(self):
+        self.order_id = uuid.uuid4()
+        self.order_date = date.today()
+        self.client_id = uuid.uuid4()
+        self.Goods = {}
```
Comment (ahodkov) [inline tasks/task07/task07.py]: List instead of dict and use `price`

File: tasks/task07/task07.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+class Good():
+  def __init__(self, name: str, price: float):
+      self.name = name
+      self.id = uuid.uuid4()
+      self.price = price
+
+
+class Order():
+    def __init__(self):
```
Comment (ahodkov) [inline tasks/task07/task07.py]: You need add parameters like in `Good`

File: tasks/task07/task07.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    @property
+    def price(self):
+      price = 0
+      if self.Goods is None or len(self.Goods.items()) < 1:
+        return price
+      for k,v  in self.Goods.items():
+        if not type(k) is Good:
+          return price
+        price += v
+      return price
```
Comment (ahodkov) [inline tasks/task07/task07.py]: `return sum(g.price for g in self.goods)`

File: tasks/task07/task07.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+          return order
+        return None
+    
+    def delete(self, order_id: uuid.UUID):
+       for order in self.items:
+          if order.order_id == order_id:
+            self.items.remove(order)
+
+    def list(self, n_latest : int = None):
+      if n_latest == None:
```
Comment (ahodkov) [inline tasks/task07/task07.py]: `if n_latest == None or n_latest >= len(self.items):`

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: tasks/task07/task07.py (hunk 4 above)
Comment (KaplenkoAnastasiia) [reply tasks/task07/task07.py]: what parameters should be added?

File: tasks/task07/task07.py (hunk 3 above)
Comment (KaplenkoAnastasiia) [reply tasks/task07/task07.py]: ok

**Review by KaplenkoAnastasiia — COMMENTED** [review]
new changes were made

File: tasks/task07/task07.py (hunk 4 above)
Comment (ahodkov) [reply tasks/task07/task07.py]: `order_id: uuid, order_date, client_id, goods: list, price = 0`
`self.order_id = order_id  self.order_date = order_date self.client_id = client_id self.Goods = goods self.price = price`

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: tasks/task07/task07.py (hunk 4 above)
Comment (KaplenkoAnastasiia) [reply tasks/task07/task07.py]: added

**Review by ahodkov — APPROVED** [review]

**Pull request merged**